src/realm/table.cpp

```diff
@@ -5970,7 +5980,14 @@ void Table::generate_patch(const Table* table, std::unique_ptr<HandoverPatch>& p
 TableRef Table::create_from_and_consume_patch(std::unique_ptr<HandoverPatch>& patch, Group& group)
 {
     if (patch) {
-        TableRef result(group.get_table(patch->m_table_num));
+        TableRef result;
+        if (patch->m_is_sub_table) {
+            auto parent_table = group.get_table(patch->m_table_num);
```

Comment:
Same question as for generate_patch: Don't you need to handle subtables of subtables etc?